config/config.go

```diff
@@ -279,3 +282,13 @@ func (req *WorkflowRestrictions) MustNotChangeSpec() bool {
 	}
 	return req.TemplateReferencing == TemplateReferencingSecure
 }
+
+// ProgressConfig defines the configuration for self reporting progress
+type ProgressConfig struct {
+	// Enabled controls whether self reporting progress is allowed or not. Default is true, set "enabled: false" to turn off
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
does not need to be pointer if default is 'false'

Reply:
default is currently true - should I make it false? Would make sense to me as progress needs to be explicitly supported by the workflow pods anyway.